### DISCUSSION TITLE
Doesn't seem to work with 3rd party mongodb datasource

### DISCUSSION THREAD

**Discussion by ycsin** [marcusolsson/grafana-hourly-heatmap-panel#35] [Q&A]
Hi, is there a way for this to work with 3rd party mongodb datasource like [mongodb-grafana-backend](https://github.com/PhracturedBlue/mongodb-grafana-backend)?

**Comment by marcusolsson** [comment]
I'm not familiar with the MongoDB data source, but I'm pretty certain it's possible. What errors are you getting?

**Comment by ycsin** [comment]
To provide more context:
I have a table that grabs (full) data from mongodb through that 3rd-party datasource plugin, I'm trying to use the hourly heatmap to visualize the counts of a key equal to a value, so I used the transform's `filter by name` to filter out the key and also the date, then followed by `filter by values`.

On the built-in `heatmap` plugin, it is displaying something:
![image](https://user-images.githubusercontent.com/1563974/123566642-c1e50000-d7f2-11eb-81a8-b3cada4317c6.png)

But on the hourly heatmap it's:
![image](https://user-images.githubusercontent.com/1563974/123566684-d88b5700-d7f2-11eb-8c2d-1106c3cafe60.png)

And nothing errors visually

**Reply by marcusolsson** [reply]
Could you check the Query inspector and let me know how the DataFrame structure looks like?

**Reply by ycsin** [reply]
This is the full output from here
![image](https://user-images.githubusercontent.com/1563974/124379458-cbd29b80-dce9-11eb-99a2-c5743d7d68a8.png)

[QueryInspector_RAW.txt](https://github.com/marcusolsson/grafana-hourly-heatmap-panel/files/6759633/QueryInspector_RAW.txt)

**Reply by marcusolsson** [reply]
Sorry for the late response. Instead of the query, could you post the value of the field in this screenshot?:

![Screenshot 2021-09-01 at 18 27 46@2x](https://user-images.githubusercontent.com/8396880/131708523-07a45571-8dbe-4692-9203-d28d0cfaabda.png)

**Reply by ycsin** [reply]
No worries.
Here it is: [dataframe.txt](https://github.com/marcusolsson/grafana-hourly-heatmap-panel/files/7112970/dataframe.txt)

I'm trying to visualize the occurrence when the `smp` equals to some value.
The dataframe above is a full query in one of the panel, then other panel will use that panel as a datasource, and filter out required data.

This is the filtered out data for this heatmap: 
[dataframe  - smp.txt](https://github.com/marcusolsson/grafana-hourly-heatmap-panel/files/7112989/dataframe.-.smp.txt)
